forge-game/src/main/java/forge/game/trigger/TriggerSpellAbilityCastOrCopy.java

```diff
@@ -306,10 +305,14 @@ public final void setTriggeringObjects(final SpellAbility sa, Map<AbilityKey, Ob
         sa.setTriggeringObject(AbilityKey.Card, castSA.getHostCard());
         sa.setTriggeringObject(AbilityKey.SpellAbility, castSA.copy(castSA.getHostCard(), true));
         sa.setTriggeringObject(AbilityKey.StackInstance, si);
-        if (!saForTargets.getTargets().isEmpty()) {
-            sa.setTriggeringObject(AbilityKey.SpellAbilityTarget, saForTargets.getTargets().get(0));
+        final List<TargetChoices> allTgts = saForTargets.getAllTargetChoices();
```

Comment:
This needs to be an FCollection to skip duplicates

Reply:
Seems I can't do an `FCollection<TargetChoices>` ?

Reply:
It should probably be on the GameEntity level list though, since `getAllTargetChoices()` basically returns a List of Lists

otherwise some duplicates might still slip through